### PR TITLE
Bump `what4` version to `1.4` in preparation for release

### PR DESCRIPTION
**Pull request by RyanGlScott**

### CODE REVIEW

File: what4/CHANGES.md [hunk 1]
```diff
@@ -9,14 +9,27 @@
 
 * Remove a dependency on `data-binary-ieee754`, which has been deprecated.
 
-* Deprecated `allSupported` which represents the SMT logic `ALL_SUPPORTED`,
+* Deprecate `allSupported` which represents the SMT logic `ALL_SUPPORTED`,
   and added `allLogic` instead which represents the SMTLib standard logic `ALL`.
```
Comment (kquick) [inline what4/CHANGES.md:13]: If you are switching to present tense, change "added" to "add"

File: what4/what4.cabal [hunk 2]
```diff
@@ -1,6 +1,6 @@
 Cabal-version: 2.4
 Name:          what4
-Version:       1.3.0.0.99
+Version:       1.4
 Author:        Galois Inc.
 Maintainer:    jhendrix@galois.com, rdockins@galois.com
 Copyright:     (c) Galois, Inc 2014-2022
```
Comment (kquick) [inline what4/what4.cabal:6]: Let's bump the copyright and add you and I (or anyone else you can think of) as Maintainer.

**Review by kquick — APPROVED** [review]
Just a couple of small fixups recommended.

File: what4/CHANGES.md (hunk 1 above)
Comment (RyanGlScott) [reply what4/CHANGES.md]: Oops, good catch.

File: what4/what4.cabal (hunk 2 above)
Comment (RyanGlScott) [reply what4/what4.cabal]: I have added us as maintainers, and I have also updated the copyright years to 2023.

**Pull request merged**